### PR TITLE
feat: add basic nova search

### LINKED ISSUE #927: [Task Tracker]: Add search
### Task description

TBD

### Requirements

TBD

### Acceptance criteria

TBD

### Creation checklist

- [ ] I have assigned this task to the correct people
- [ ] I have added the most appropriate labels
- [ ] I have linked the correct milestone and/or project

### PR DESCRIPTION
**Pull request by brancoder**
# Description of change

resolves #927

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## Change checklist

- [x] My code follows the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: api/src/utils/nova/addressHelper.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    AnchorAddress,
+    Utils,
+    ImplicitAccountCreationAddress,
+    RestrictedAddress,
+} from "@iota/sdk-nova";
+import { plainToInstance } from "class-transformer";
+import { IAddressDetails } from "../../models/api/nova/IAddressDetails";
+import { HexHelper } from "../hexHelper";
+
+export class AddressHelper {
```
Comment (begonaalvarezd) [inline api/src/utils/nova/addressHelper.ts:19]: this whole thing is duplicated with the client one right? lets add a TODO and create a task so we try to improve this, we should have shared/common code we can reuse between both modules

File: api/src/utils/nova/searchQueryBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+            // also perform a tag search
+            const maybeTag = Converter.isHex(this.query, true) ? HexHelper.addPrefix(this.query) : Converter.utf8ToHex(this.query, true);
+            if (maybeTag.length < 66) {
+                tag = maybeTag;
+            }
+        }
+
+        return {
+            queryLower: this.queryLower,
```
Comment (begonaalvarezd) [inline api/src/utils/nova/searchQueryBuilder.ts:142]: if its an integer, we could redirect to `slot`, we can treat them in the same way we treat milestones now in stardust and chrysalis

**Review by begonaalvarezd — COMMENTED** [review]
amazig work 🚀 
didnt finish the review so for now just added 2 random comments 🙏🏼

File: api/src/utils/nova/searchQueryBuilder.ts (hunk 2 above)
Comment (brancoder) [reply api/src/utils/nova/searchQueryBuilder.ts:148]: Sure :+1:  But we will also need a new task to implement the search improvements for slots, epochs, delegation...

File: api/src/utils/nova/addressHelper.ts (hunk 1 above)
Comment (brancoder) [reply api/src/utils/nova/addressHelper.ts:19]: Created :point_right: : https://github.com/iotaledger/explorer/issues/1116

File: api/src/utils/nova/searchQueryBuilder.ts (hunk 2 above)
Comment (brancoder) [reply api/src/utils/nova/searchQueryBuilder.ts:148]: Created the task :point_right: https://github.com/iotaledger/explorer/issues/1117

**Review by begonaalvarezd — APPROVED** [review]

**Pull request merged**